tools/clang/test/HLSL/workgraph/attribute_diags.hlsl

```diff
@@ -0,0 +1,130 @@
+// RUN: %clang_cc1 -fsyntax-only -verify %s
+
+// NodeDispatchGrid and NodeMaxDispatchGrid may only be used with Broadcasting launch nodes
+
+[Shader("node")]
+[NodeLaunch("Coalescing")]           // expected-note +{{Launch type defined here}}
+[NodeDispatchGrid(4, 4, 2)]          // expected-error {{'nodedispatchgrid' may only be used with broadcasting nodes}}
+[NumThreads(32, 1, 1)]
+void node01()
+{ }
+
+[Shader("node")]
+[NodeLaunch("Thread")]               // expected-note +{{Launch type defined here}}
+[NodeDispatchGrid(8, 4, 2)]          // expected-error {{'nodedispatchgrid' may only be used with broadcasting nodes}}
+void node02()
+{ }
+
+[Shader("node")]
+[NodeLaunch("Coalescing")]           // expected-note +{{Launch type defined here}}
+[NodeMaxDispatchGrid(8, 8, 8)]       // expected-error {{'nodemaxdispatchgrid' may only be used with broadcasting nodes}}
+[NumThreads(32, 1, 1)]
+void node03()
+{ }
+
+[Shader("node")]
+[NodeLaunch("Thread")]               // expected-note +{{Launch type defined here}}
+[NodeMaxDispatchGrid(256, 8, 8)]     // expected-error {{'nodemaxdispatchgrid' may only be used with broadcasting nodes}}
+void node04()
+{ }
+
+// NodeDispatchGrid and NodeMaxDispatchGrid may not both be used
+
+[Shader("node")]
+[NodeLaunch("Broadcasting")]
+[NodeMaxDispatchGrid(256, 64, 32)]   // expected-error {{Node shader 'node05' may not use both 'nodemaxdispatchgrid' and 'nodedispatchgrid'}}
+[NodeDispatchGrid(32, 1, 1)]         // expected-note  {{nodedispatchgrid defined here}}
```

Comment:
Seems a bit odd that the second attribute is the one that gets the note. Any idea why that is? Maybe we need a deterministic container somewhere?

Reply:
It appears that the attributes are processed in reverse order. I don't know whether there is a Clang reason to do it that way, or what the implications of changing it would be - I'll investigate.

Reply:
Except that's not true of the NodeLaunch attrib above 🤔

Reply:
Scratch that, the above is a different situation. This may well be alphabetical. Anyway, I'm not too concerned about it just now